File: Java/Chapter12/FactoryExample.java
public class FactoryExample {

    public static void main(String args[]) {
	ITv tv;
	String tvSpec = args[0];
	String region = args[1];
	int channel = Integer.parseInt(args[2]);

	try {
	    tv = TvFactory.createTv(tvSpec);
	    tv.setRegion(region);
	    tv.display(channel);
	} catch (TvException te) {
	    System.out.println(te.getMessage());
	}
    }

}

File: Python/Chapter05/factory_example.py
import sys
from tv_factory import TvFactory
from tv_exception import TvException

def main(args):
    tvSpec = args[1]
    region = args[2]
    channel = int(args[3])
    
    try:
        tv = TvFactory.createtv(tvSpec)
        tv.region = region
        tv.display(channel)
    except TvException as te:
        print(te)
        
if __name__ == '__main__':
    main(sys.argv)
